components/brave_rewards/browser/publisher_info_database.cc

```diff
@@ -1106,6 +1089,99 @@ bool PublisherInfoDatabase::MigrateV4toV5() {
   return transaction.Commit();
 }
 
+bool PublisherInfoDatabase::MigrateV5toV6() {
+  sql::Transaction transaction(&GetDB());
+  if (!transaction.Begin()) {
+    return false;
+  }
+
+  const char* activity = "activity_info";
+  if (GetDB().DoesTableExist(activity)) {
+    std::string sql = "ALTER TABLE activity_info RENAME TO activity_info_old;";
+
+    if (!GetDB().Execute(sql.c_str())) {
+      return false;
+    }
+
+    if (!CreateActivityInfoTable()) {
+      return false;
+    }
+
+    if (!CreateActivityInfoIndex()) {
+      return false;
+    }
+
+    const std::string columns_insert = "publisher_id, "
+                                       "duration, "
+                                       "visits, "
+                                       "score, "
+                                       "percent, "
+                                       "weight, "
+                                       "reconcile_stamp";
+
+    const std::string columns_select = "publisher_id, "
+                                       "sum(duration) as duration, "
+                                       "sum(visits) as duration, "
+                                       "sum(score) as score, "
+                                       "percent, "
+                                       "weight, "
+                                       "reconcile_stamp";
+
+    sql = "PRAGMA foreign_keys=off;";
+    sql.append("INSERT INTO activity_info (" + columns_insert + ") "
+               "SELECT " + columns_select + " "
+               "FROM activity_info_old "
+               "GROUP BY publisher_id, reconcile_stamp;");
+    sql.append("DROP TABLE activity_info_old;");
+    sql.append("PRAGMA foreign_keys=on;");
+
+    bool result = GetDB().Execute(sql.c_str());
+
+    if (!result) {
+      LOG(ERROR) << "DB: Error with MigrateV5toV6";
```

Comment:
If there is an error should we not `return false`?

Reply:
we have this queries in transaction, so transaction at the end will fail

Reply:
what would you suggest to do in this case?

Reply:
we have all of this code wrapped in transaction, so transaction will fail. I just added this log as I needed to used return value from Execute. But now that you pointed out I think we shouldn't increase db number if migration fail

Reply:
done